src/firmware/types.rs

```diff
@@ -202,3 +210,512 @@ pub struct SnpPlatformStatus {
     /// Reported TCB version.
     pub reported_tcb_version: TcbVersion,
 }
+
+#[repr(C)]
+pub struct SnpConfig {
+    reserved: [u8; 52],
+}
+
+impl Default for SnpConfig {
+    fn default() -> Self {
+        Self { reserved: [0; 52] }
+    }
+}
+
+#[repr(C)]
+pub struct SnpSetExtConfig<'a> {
+    /// Address of the SnpConfig or 0 when reported_tcb does not need
+    /// to be updated.
+    pub config_address: Option<&'a SnpConfig>,
+
+    /// Address of extended guest request certificate chain or 0 when
+    /// previous certificate should be removed on SNP_SET_EXT_CONFIG.
+    pub certs_address: Option<*mut c_void>,
+
+    /// Length of the certificates.
+    pub certs_len: u32,
+}
+
+#[repr(C)]
+pub struct SnpGetExtConfig<'a> {
+    /// Address of the SnpConfig or 0 when reported_tcb does not need
+    /// to be updated.
```

Comment:
Pulled the comments from [sev-snp-part2-v6](https://github.com/AMDESE/linux/blob/sev-snp-part2-v6/include/uapi/linux/psp-sev.h). I am open to discussion about the wording, though.

Reply:
Yes. The kernel code uses the same struct for both SET and GET commands, but the field comments are explaining the SET usage (that's why I didn't comment on `SnpSetExtConfig` above).

Since you have two different structs here, we can make the SnpGetExtConfig explain the correct usage of the fields in the GET request (that is, if you pass in 0, you ask the kernel to NOT fetch that info).

Reply:
Gotcha, I understand what you are saying now. I will make the adjustment :+1: